publish.js

```diff
@@ -105,53 +108,48 @@ function patchedManifest (spec, auth, base, opts) {
   return manifest
 }
 
-function buildMetadata (spec, auth, registry, manifest, tardata, opts) {
+function buildMetadata (auth, registry, manifest, tarballData, opts) {
+  const { access, defaultTag, algorithms } = opts
   const root = {
     _id: manifest.name,
     name: manifest.name,
     description: manifest.description,
     'dist-tags': {},
     versions: {},
+    access,
     readme: manifest.readme || ''
   }
 
-  if (opts.access) root.access = opts.access
-
-  if (!auth.token) {
-    root.maintainers = [{ name: auth.username, email: auth.email }]
-    manifest.maintainers = JSON.parse(JSON.stringify(root.maintainers))
-  }
-
   root.versions[manifest.version] = manifest
-  const tag = manifest.tag || opts.tag
+  const tag = manifest.tag || defaultTag
```

Comment:
This was brand new news to me.  I'm going to go document it in `npm help package.json` right now, actually.